django/contrib/postgres/fields/serial.py

```diff
@@ -0,0 +1,98 @@
+from django.core import checks
+from django.db import models
+from django.db.models import NOT_PROVIDED
+from django.db.models.expressions import DatabaseDefault
+from django.utils.translation import gettext_lazy as _
+
+__all__ = ("BigSerialField", "SmallSerialField", "SerialField")
+
+
+class SerialFieldMixin:
+    db_returning = True
+
+    def __init__(self, *args, **kwargs):
+        kwargs["blank"] = True
+        super().__init__(*args, **kwargs)
+
+    def deconstruct(self):
+        name, path, args, kwargs = super().deconstruct()
+        kwargs.pop("blank")
+        return name, path, args, kwargs
+
+    def check(self, **kwargs):
+        return [
+            *super().check(**kwargs),
+            *self._check_not_null(),
+            *self._check_default(),
+            *self._check_database_default(),
+        ]
+
+    def _check_not_null(self):
+        if self.null:
+            return [
+                checks.Error(
+                    "SerialFields do not accept null values.",
```

Comment:
```suggestion
                f"{self.__class__.__name__} must not have ``null=True``.",
```
Saying "must not have `null=True`" aligns more to `fields.E007`.

Also this is currently documented as 
```
``<field>`` does not accept null values.
```
But always says `SerialFields`, maybe we should do something similar to `fields.E010` and update the error message depending on the field. If so, update each of these, otherwise I would document the check to be:
```
SerialFields does not accept null values.
```

Reply:
@sarahboyce , I was following `fields.E110` which has:
```
``BooleanField``\s do not accept null values
```
It was @ngnpope 's idea to make this error message general, and I agree with it, because this way we wouldn't need to define a new error code every time a new field doesn't accept a `null` value.